drivers/dma/dma_stm32.c

```diff
@@ -620,11 +620,34 @@ static int dma_stm32_init(struct device *dev)
 	return 0;
 }
 
+static int dma_stm32_get_status(struct device *dev, u32_t id,
+				struct dma_status *stat)
+{
+	const struct dma_stm32_config *config = dev->config_info;
+	DMA_TypeDef *dma = (DMA_TypeDef *)(config->base);
+	struct dma_stm32_data *data = dev->driver_data;
+
+	/* give channel from index 0 */
+	id = id - STREAM_OFFSET;
+	if (id >= data->max_streams) {
+		return -EINVAL;
+	}
+
+	struct dma_stm32_stream *stream = &data->streams[id - STREAM_OFFSET];
```

Comment:
id is id - STREAM_OFFSET now.

Reply:
You are right. Sorry for rushing to a new mistake. :(